### DISCUSSION TITLE
Cefsharp (Libcef.dll) program crashes after opening web page from WPF

### DISCUSSION THREAD

**Discussion by US-CPHAM** [cefsharp/CefSharp#4954] [General]
Hello everyone. I am having an issue with using CefSharp to open a web page from a WPF application.

In WPF, I open a window, and in that window, I use ChromiumWebBrowser (CefSharp) to navigate to a web page. When I run it in the Debug environment, there are no issues, and everything runs smoothly. However, when I build the application and run it in the production environment, the program crashes a few seconds after the website screen is displayed. Checking the Windows Event log, I get the following result:

Faulting application name: Test.exe, version: 6.0.0.135, time stamp: 0x6707d47d
Faulting module name: libcef.dll, version: 129.0.11.0, time stamp: 0x66fb0c5a
Exception code: 0x4000001f
Error offset: 0x04ea8352
Faulting process ID: 0x0x391C
Faulting application start time: 0x0x1DB1BBC15CD5A5C
Faulting application path: C:\Program Files (x86)\Test\Test.exe
Faulting module path: C:\Program Files (x86)\Test\libcef.dll
Report identifier: a1c6a7fb-e83b-4853-b6cc-a1c05b13494c
Full name of the faulting package:
Application ID relative to the faulting package:
I don't know how to handle this, and I need help. Thanks in advance."

**Comment by amaitland** [comment]
> However, when I build the application and run it in the production environment, the program crashes a few seconds after the website screen is displayed

What's the production environment exactly?

There really isn't a lot to go on here.

Additional you can use https://github.com/cefsharp/CefSharp.MinimalExample as a reference for testing purposes.

**Comment by khcpietro** [comment]
I had a same problem, but I solved with the example project @amaitland mentioned.
`app.manifest` file is the key. More specifically you should add compatibility block like below.
Without this block, the example project also get crashed.

```
<?xml version="1.0" encoding="utf-8"?>
<asmv1:assembly
  manifestVersion="1.0"
  xmlns="urn:schemas-microsoft-com:asm.v1"
  xmlns:asmv1="urn:schemas-microsoft-com:asm.v1"
  xmlns:asmv2="urn:schemas-microsoft-com:asm.v2"
  xmlns:asmv3="urn:schemas-microsoft-com:asm.v3"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

	<compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
		<application>
			<!-- A list of all Windows versions that this application is designed to work with. 
      Windows will automatically select the most compatible environment.-->

			<!-- Windows 7 -->
			<supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
			<!-- Windows 8 -->
			<supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
			<!-- Windows 8.1 -->
			<supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
			<!-- Windows 10 -->
			<supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}" />
		</application>
	</compatibility>
</asmv1:assembly>
```

One more thing, you should add this line to `csproj` file inside `<PropertyGroup>`.

`<ApplicationManifest>app.manifest</ApplicationManifest>`
